### ISSUE TITLE
Seemingly random struct.error error exception

### ISSUE THREAD

**Issue by WillCodeForCats** [WillCodeForCats/solaredge-modbus-multi#424]
### Describe the bug

Seemingly random `struct.error` error exception. Maybe caused by an incomplete or short response to the expected read?

### Expected behavior

Exception shouldn't happen.

### Diagnostic File

N/A (observed in dev environment with simulator)

### Debug logs

```shell
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 283, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solaredge_modbus_multi/__init__.py", line 191, in _async_update_data
    return await self._refresh_modbus_data_with_retry(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solaredge_modbus_multi/__init__.py", line 229, in _refresh_modbus_data_with_retry
    raise ex
  File "/config/custom_components/solaredge_modbus_multi/__init__.py", line 226, in _refresh_modbus_data_with_retry
    return await self._hub.async_refresh_modbus_data()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/solaredge_modbus_multi/hub.py", line 410, in async_refresh_modbus_data
    await inverter.read_modbus_data()
  File "/config/custom_components/solaredge_modbus_multi/hub.py", line 835, in read_modbus_data
    ("AC_Current_B", decoder.decode_16bit_uint()),
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pymodbus/payload.py", line 375, in decode_16bit_uint
    return unpack(fstring, handle)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^
struct.error: unpack requires a buffer of 2 bytes
```

### Home Assistant Version

2023.8.4

### solaredge-modbus-multi Version

pymodbus_3_4_1 branch

### Installation Type

HAOS

### Read the Instructions

- [X] I have read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki
- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration

### Additional Context

Observed randomly, usually after sending commands.

**Comment by WillCodeForCats** [comment]
PR #426 may fix this, but it is extremely difficult to duplicate.

**Comment by WillCodeForCats** [comment]
Marking as resolved for Release v2.4.4
